src/librepgp/stream-armor.c

```diff
@@ -368,10 +374,17 @@ armored_src_read(pgp_source_t *src, void *buf, size_t len)
             *bptr++ = (*dptr << 2) | (*(dptr + 1) >> 4);
         }
 
-        param->crc = crc24_update(param->crc, param->rest, bptr - param->rest);
+        uint8_t    crc_fin[5];
+        pgp_hash_t fin_ctx = {0};
+
+        /* Calculate CRC after reading whole input stream */
+        pgp_hash_add(&param->crc_ctx, param->rest, bptr - param->rest);
+        if (!pgp_hash_copy(&fin_ctx, &param->crc_ctx) || !pgp_hash_finish(&fin_ctx, crc_fin)) {
```

Comment:
Actually that's good question. I'm not sure if I can use context after it has been finalised. To be checked.

Reply:
So ``pgp_hash_finish`` will distroy context - so it must be cloned as it may be used later.
But if I call botan directly, then I think I may be able to reuse the context.

Reply:
Hm, where it may be used later? Once all input is read and crc validated no other actions will be taken on it.

Reply:
Ah, there is a assumption that ``armored_src_read`` is not called again. Not sure I like the assumption, but we could also simply finalize here.

Reply:
Does it make more sense if we calculate it in "armor_src_close" (after changing the function so that it returns an error code)?

Reply:
Ehh... everything what I wrote is wrong and I'm stupid.
Thanks, to ``eofb64`` the context finalization will be done only once, right?

Reply:
@flowher Right, eofb64 == true happens only once, when CRC or trailer '=' are found on the input stream. After that there is no decoding.

Reply:
@ni4 Perfect! Thank you, very useful.